src/escape/rooms/andrewRoom.java

```diff
@@ -0,0 +1,232 @@
+
+          		}else {
+          			shownMap [priorX][priorY] = "-";
+              		shownMap [priorX - 3][priorY] = "X";
+              		priorX -= 3;
+          		}
+          		if(cavernMap[priorX + 1][priorY] != " " ) {
+          			shownMap[priorX + 1][priorY] = cavernMap[priorX + 1][priorY];
+          		}
+          	}
+          	if(move == "S" ) {
+          		
+          		if(priorY == 9) {
+          			shownMap [priorX][priorY] = "-";
+          			shownMap [priorX][9] = "X";
+          			priorY = 0;
+
+          		}else {
+          			shownMap [priorX][priorY] = "-";
+              		shownMap [priorX][priorY + 1] = "X";
+              		priorY++;
+          		}
+          		if(cavernMap[priorX + 1][priorY] != " " ) {
+          			shownMap[priorX + 1][priorY] = cavernMap[priorX + 1][priorY];
+          		}
+          	}
+          	if(move == "D" ) {
+          		if(priorX == 22) {
+          			shownMap [priorX][priorY] = "-";
+          			shownMap [1][priorY] = "X";
+          			priorX = 1;
+
+          		}else {
+          			shownMap [priorX][priorY] = "-";
+              		shownMap [priorX + 3][priorY] = "X";
+              		priorX += 3;
+          		}
+          		if(cavernMap[priorX + 1][priorY] != " " ) {
+          			shownMap[priorX + 1][priorY] = cavernMap[priorX + 1][priorY];
+          		}
+          	}
+          	if(move == "x") {
+          		if(priorX == 19 && priorY == 4) {
+          			System.out.println("You win and can move on to the next room");
+          			moveOn = true;
+          		}else {
+          			System.out.println("You missed.");
+          			System.out.println("Resetting room...");
+          			 try {
+          					Thread.sleep(5000);
+          				} catch (InterruptedException e) {
+          					// TODO Auto-generated catch block
+          					e.printStackTrace();
+          				}
+          		        }
+          		}
+          		
+          	printArray(shownMap);
+          	}
+          	
+    	  
+          	
+          	if(priorX == 20 && priorY == 4) {
+          		end = true;
+          		System.out.println("The wumpus ate you");
+          		System.out.println("Resetting room...");
+          	}
+          	if(priorX == 5 && priorY == 8) {
+          		end = true;
+          		System.out.println("You fell in a slime pit");
+          		System.out.println("Resetting room...");
+          	}
+        		
+          
+          //Print full map
+          	printArray(cavernMap);
+    	  
+      }
+        
+        
+		
+        try {
+			Thread.sleep(10000);
+		} catch (InterruptedException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+        }
+    //not quite working
+    //TODO fix printing Arrays
+    public void printArray(String[][] array) {
+    	int i,j;
+    	for(i = 0; i < shownMap.length; i++) {
+    		for(j = 0; j < shownMap[1].length; j++) {
+    			if(array == cavernMap) {
+    				System.out.println(cavernMap[j][i] + " ");
```

Comment:
System.out.println will enter a space after each character. You most likely want to just have a System.out.print with a println or return character printed after every time the first for loop finishes.